src/cljs/rems/user_settings.cljs

```diff
@@ -7,8 +7,16 @@
 (rf/reg-event-fx
  :loaded-user-settings
  (fn [{:keys [db]} [_ user-settings]]
-   (language/update-language (:language user-settings))
-   {:db (assoc db :user-settings user-settings)}))
+   (let [new-language (or (language/get-language-cookie)
+                          (:language user-settings))
+         new-settings (assoc user-settings :language new-language)]
+     (language/update-language new-language)
+     ;; if the user has changed language before login
+     ;; the cookie will not match the language setting
+     ;; and we should save the new language setting
+     (when-not (= (language/get-language-cookie) (:language user-settings))
+       (rf/dispatch [::update-user-settings new-settings]))
```

Comment:
Isn't this called also when the user has no cookie?

Reply:
Yes it is. Now it's slightly different.